mlflow/server/js/src/model-registry/components/SchemaTable.tsx

```diff
@@ -54,16 +54,44 @@ export class SchemaTableImpl extends React.PureComponent<Props> {
     );
   };
 
-  getSchemaTypeRepr = (schemaTypeSpec: any) => {
-    let { type } = schemaTypeSpec;
-    if (schemaTypeSpec.type === MODEL_SCHEMA_TENSOR_TYPE) {
-      type = `Tensor (dtype: ${schemaTypeSpec['tensor-spec'].dtype}, shape: [${schemaTypeSpec['tensor-spec'].shape}])`;
+  getSchemaTypeRepr = (schemaTypeSpec: ColumnSpec | TensorSpec): string => {
+    const { type } = schemaTypeSpec;
+
+    let repr: string = type;
+    if (schemaTypeSpec.type === 'tensor') {
+      repr = `Tensor (dtype: ${schemaTypeSpec['tensor-spec'].dtype}, shape: [${schemaTypeSpec['tensor-spec'].shape}])`;
+    } else {
+      repr = this.getColumnTypeRepr(schemaTypeSpec);
     }
 
     // If the "optional" property is present and true, wrap the type around an "Optional[]"
+    // NOTE: newer model signatures will have "required" instead of "optional", but we currently
+    // have to support both since both exist.
     if (schemaTypeSpec.optional) {
-      type = `Optional[${type}]`;
+      repr = `Optional[${repr}]`;
+    } else if (schemaTypeSpec.required) {
+      repr = `${repr} (required)`;
     }
```

Comment:
Could we switch the order of these two condition checks? Just let `required` be considered first.